Simple nft example: testnet opensea -> metamask extension

rin-st opened the pull request:
note: I added images to `public` folder and used urls like `https://raw.githubusercontent.com/scaffold-eth/se-2-challenges/challenge-simple-nft-example/extension/packages/nextjs/public/nfts.png` (not working until merge to `challenge-simple-nft-example` branch), since gh doesn't create assets like `https://github.com/scaffold-eth/se-2-challenges/assets/...` anymore. When drag'n'drop image to the message gh creates link like `https://github.com/user-attachments/assets/...`, but these links are not very stable, so using images from the public folder

@escottalexander, you'll need to resolve conflicts if this PR is merged earlier than #361 

Fixes #347

escottalexander reviewed (approved):
This is very simple and clear. Good work.

rin-st commented:
Changed target branch to `challenge-tokenization`, fixed urls, conflicts and tested installation with this branch. Works as expected. Merging, thanks!
